src/link.rs

```diff
@@ -20,45 +20,27 @@
 
 #[cfg(feature="runtime")]
 macro_rules! link {
-    (@IMPL: #[cfg($cfg:meta)] fn $name:ident($($pname:ident: $pty:ty), *) $(-> $ret:ty)*) => (
+    (@LOAD: #[cfg($cfg:meta)] fn $name:ident($($pname:ident: $pty:ty), *) $(-> $ret:ty)*) => (
         #[cfg($cfg)]
-        pub fn $name(library: &mut super::SharedLibrary, allow_failure: bool) -> Result<(), String> {
-            let symbol = unsafe { library.library.get(stringify!($name).as_bytes()) }.map_err(|_| {
-                format!("could not load `{}`", stringify!($name))
-            });
-            let symbol = if allow_failure {
-                symbol.ok().map(|s| *s)
-            } else {
-                Some(*try!(symbol))
-            };
-            library.functions.$name = symbol;
-            Ok(())
+        pub fn $name(library: &mut super::SharedLibrary) {
+            let symbol = unsafe { library.library.get(stringify!($name).as_bytes()) }.ok();
+            library.functions.$name = symbol.map(|s| *s);
         }
 
         #[cfg(not($cfg))]
-        pub fn $name(_: &mut super::SharedLibrary, _: bool) -> Result<(), String> {
-            Ok(())
-        }
+        pub fn $name(_: &mut super::SharedLibrary) {}
     );
 
-    (@IMPL: fn $name:ident($($pname:ident: $pty:ty), *) $(-> $ret:ty)*) => (
-        pub fn $name(library: &mut super::SharedLibrary, allow_failure: bool) -> Result<(), String> {
-            let symbol = unsafe { library.library.get(stringify!($name).as_bytes()) }.map_err(|_| {
-                format!("could not load `{}`", stringify!($name))
-            });
-            let symbol = if allow_failure {
-                symbol.ok().map(|s| *s)
-            } else {
-                Some(*try!(symbol))
-            };
-            library.functions.$name = symbol;
-            Ok(())
+    (@LOAD: fn $name:ident($($pname:ident: $pty:ty), *) $(-> $ret:ty)*) => (
+        pub fn $name(library: &mut super::SharedLibrary) {
+            let symbol = unsafe { library.library.get(stringify!($name).as_bytes()) }.ok();
+            library.functions.$name = symbol.map(|s| *s);
         }
     );
 
     ($($(#[cfg($cfg:meta)])* pub fn $name:ident($($pname:ident: $pty:ty), *) $(-> $ret:ty)*;)+) => (
         use std::cell::{RefCell};
-        use std::sync::{Mutex};
+        use std::sync::{Arc, Mutex};
```

Comment:
The `Mutex` was for protecting a `thread_local!` variable that doesn't exist anymore. Could you remove the `Mutex` import?